src/components/staff-payment/StaffPayment.vue

```diff
@@ -214,17 +217,13 @@ export default class StaffPayment extends Vue {
         this.$refs.folioNumberInputRef.resetFolioNumberValidation()
         // enable validation for this form
         this.$refs.fasForm.validate()
-        // update data
-        this.emitStaffPaymentData({ option: StaffPaymentOptions.FAS })
```

Comment:
`this.emitStaffPaymentData` is now called only from `onStaffPaymentDataChanged`. This simplifies the flow a bit, as `onStaffPaymentDataChanged` will always be triggered when `onPaymentOptionChanged` changes.